src/main/java/reposense/model/Commit.java

```diff
@@ -0,0 +1,87 @@
+package reposense.model;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Represents a git commit in {@code RepoConfiguration}.
+ */
+public class Commit {
```

Comment:
This may be confused with the `CommitInfo` we are using as well.
Perhaps rename to `IgnoreCommit` or `CommitHash`, to be more specific (update the header docs as well)?